### PR TITLE
REF: implement array_algos.take

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
if you can rebase

**Comment by jbrockmendel** [comment]
rebased + green

**Comment by jreback** [comment]
maybe just more all of algorithms.py to array_algos? though i think these should probably be in somewhat separated modules rather than one giant one. so ok with moveing take but let's put it separate as its pretty meaty by itself.

**Comment by jbrockmendel** [comment]
> maybe just more all of algorithms.py to array_algos?

there are some that depend on Index/Series, so they dont all fit

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -1414,219 +1414,6 @@ def get_indexer(current_indexer, other_indexer):
...
-    def func2(arr, indexer, out, fill_value=np.nan):
-        indexer = ensure_int64(indexer)
-        _take_nd_object(
-            arr, indexer, out, axis=axis, fill_value=fill_value, mask_info=mask_info
-        )
-
-    return func2
-
-
 def take(
```
Comment (jreback) [inline pandas/core/algorithms.py:1417]: how are we not moving this?

Comment (jbrockmendel) [reply pandas/core/algorithms.py:1417]: its high-level and i dont want to futz with updating imports everywhere else

**Comment by jreback** [comment]
k thanks

File: pandas/core/array_algos/take.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+    Returns
+    -------
+    subarray : array-like
+        May be the same type as the input, or cast to an ndarray.
+    """
+    if fill_value is lib.no_default:
+        fill_value = na_value_for_dtype(arr.dtype, compat=False)
+
+    arr = extract_array(arr, extract_numpy=True)
```
Comment (jorisvandenbossche) [inline pandas/core/array_algos/take.py:66]: "array algos" should only deal with arrays IMO

Comment (jbrockmendel) [reply pandas/core/array_algos/take.py:66]: good catch, will fix in follow-up

**Pull request merged**